vllm/worker/multi_step_model_runner.py

```diff
@@ -409,6 +415,62 @@ def vocab_size(self) -> int:
         return self._base_model_runner.vocab_size
 
 
+DeferredLogprobsReturnType = Tuple[Optional[List[Optional[PromptLogprobs]]],
+                                   Optional[List[SampleLogprobs]]]
+
+
+def deferred_pythonize_logprobs(
+    output: SamplerOutput,
+    sampling_metadata: SamplingMetadata,
+) -> DeferredLogprobsReturnType:
+    '''
+    Perform deferred logprob Pythonization.
+
+    1. Pythonize GPU-side sampler result
+       tensors into CPU-side sampler result.
+    2. Pythonize GPU-side logprobs tensor into
+       CPU-side logprobs lists, utilizing 
+       the Pythonized sampler result computed
+       in step 1.
+    
+    These deferred computations are not required for
+    single-step scheduling or the `profile_run()`
+    phase of multi-step scheduling.
+
+    Arguments:
+
+    * output: sampler output (under deferred Pythonization)
+    * sampling_metadata
+
+    Returns:
+
+    * prompt_logprobs (CPU)
+    * sample_logprobs (CPU)
+    '''
```

Comment:
Please use the following docstring style in all added functions.
```suggestion
    """Perform deferred logprob Pythonization.

    1. Pythonize GPU-side sampler result tensors into CPU-side sampler result.
    2. Pythonize GPU-side logprobs tensor into CPU-side logprobs lists,
       utilizing  the Pythonized sampler result computed in step 1.
    
    These deferred computations are not required for single-step scheduling
    or the `profile_run()` phase of multi-step scheduling.

    Args:
        output: sampler output (under deferred Pythonization)
        sampling_metadata

    Returns:
        prompt_logprobs (CPU), sample_logprobs (CPU)
    """
```

Reply:
Thanks, applied this change where needed.